src/logic/safe/store/actions/transactions/fetchTransactions/loadGatewayTransactions.ts

```diff
@@ -0,0 +1,93 @@
+import axios, { AxiosResponse } from 'axios'
+
+import { getSafeClientGatewayBaseUrl } from 'src/config'
+import { HistoryGatewayResponse, QueuedGatewayResponse } from 'src/logic/safe/store/models/types/gateway'
+import { checksumAddress } from 'src/utils/checksumAddress'
+
+/*************/
+/*  HISTORY  */
+/*************/
+const getHistoryTransactionsUrl = (safeAddress: string): string => {
+  const address = checksumAddress(safeAddress)
+  return `${getSafeClientGatewayBaseUrl(address)}/transactions/history/`
+}
+
+const historyPointers: { [safeAddress: string]: { next: string | null; previous: string | null } } = {}
```

Comment:
if the user changes the safe, these pointers should be discarded, because if he comes back, they should see the top of the list.

I suggest persisting the pointers only for the current safe.

Reply:
+1